src/features/user/userThunk.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  AUTH_ME_ENDPOINT,
+  AUTH_REMOVE_ACCOUNT_ENDPOINT,
+  AUTH_REMOVE_PHOTO_ENDPOINT,
+  AUTH_UPDATE_PASSWORD_ENDPOINT,
+  AUTH_UPLOAD_PHOTO_ENDPOINT,
+  UPLOADS_BASE_URL,
+} from "@/constants/api"
+import axiosProtectedInstance from "@/libs/axios/axiosPrivateInstance"
+import { ChangePasswordFormValues } from "@/libs/yup/schemas/changePassword"
+import thunkErrorHandler from "@/utils/thunkErrorHandler"
+import { createAsyncThunk } from "@reduxjs/toolkit"
+import { toast } from "react-toastify"
+import { LogoutUserReason, logoutUser } from "../auth/authSlice"
+import { AvatarFormValues } from "@/libs/yup/schemas/avatar"
+import { RootState } from "@/app/store"
+import { ChangeUserDataBody } from "@/types/api"
+
+export const getUserData = createAsyncThunk(
+  "user/getUserData",
+  async (_, thunkAPI) => {
+    try {
+      const { data } = await axiosProtectedInstance.get(AUTH_ME_ENDPOINT)
+      return data
+    } catch (error) {
+      return thunkErrorHandler({ error, thunkAPI })
+    }
+  },
+)
+
+export const changeUserData = createAsyncThunk(
+  "user/changeUserData",
+  async (
+    { firstName, lastName, email, photo }: ChangeUserDataBody,
+    thunkAPI,
+  ) => {
+    try {
+      const requestData = photo
```

Comment:
You can shorten this alil by passing the obj key only if it exists

 const requestData = {
        firstName,
        lastName,
        email,
        ...(photo && { photo }),
      }